### PR TITLE
Add PortReuseMode to CosmosClientOptions

### PR DESCRIPTION
**Pull request by ausfeldt**
## Description

Add PortReuseMode to CosmosClientOptions.

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/PortReuseMode.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+//------------------------------------------------------------
+namespace Microsoft.Azure.Cosmos
+{
+    /// <summary>
+    /// Port reuse policy options used by the transport stack
+    /// </summary>
+    public enum PortReuseMode
+    {
+        /// <summary>
+        /// Reuse unicast port
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/PortReuseMode.cs]: Do we have any more info on when someone should use one vs the other?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/CosmosClientOptionsUnitTests.cs [hunk 2]
```diff
@@ -142,13 +144,15 @@ public void VerifyCosmosConfigurationPropertiesGetUpdated()
...
             Assert.AreEqual(maxTcpConnectionsPerEndpoint, clientOptions.MaxTcpConnectionsPerEndpoint);
+            Assert.AreEqual(portReuseMode, clientOptions.PortReuseMode);
 
             //Verify GetConnectionPolicy returns the correct values
             policy = clientOptions.GetConnectionPolicy();
             Assert.AreEqual(idleTcpConnectionTimeout, policy.IdleTcpConnectionTimeout);
             Assert.AreEqual(openTcpConnectionTimeout, policy.OpenTcpConnectionTimeout);
             Assert.AreEqual(maxRequestsPerTcpConnection, policy.MaxRequestsPerTcpConnection);
             Assert.AreEqual(maxTcpConnectionsPerEndpoint, policy.MaxTcpConnectionsPerEndpoint);
+            Assert.AreEqual(portReuseMode, policy.PortReuseMode);
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/CosmosClientOptionsUnitTests.cs]: Please add a UT verifying that the two Documents.PortReuseMode and Cosmos.PortReuseMode are always in sync.

File: changelog.md [hunk 3]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 
 - [#1000](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/1000) Add PortReuseMode to CosmosClientOptions.
+- [#995](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/995) Included session token in diagnostics
```
Comment (j82w) [inline changelog.md]: nit: reorder so PR numbers are descending

File: changelog.md.orig [hunk 4]
```diff
@@ -0,0 +1,190 @@
+# Changelog
```
Comment (j82w) [inline changelog.md.orig]: Remove this file? It looks like a merge file

Comment (ausfeldt) [reply changelog.md.orig]: oops

**Pull request merged**